src/InteractiveWindow/Editor/InteractiveWindow.cs

```diff
@@ -1418,26 +1425,33 @@ private bool CanExecuteActiveCode()
         #endregion
 
         #region Buffers, Spans and Prompts
-        private ReplSpan CreateStandardInputPrompt()
+        private ITrackingSpan CreateStandardInputPrompt()
         {
-            return CreatePrompt(string.Empty, ReplSpanKind.StandardInputPrompt, LastLineNumber);
+            return CreateTrackingSpan(_standardInputPromptBuffer, string.Empty, ReplSpanKind.StandardInputPrompt);
         }
 
-        private ReplSpan CreatePrimaryPrompt()
+        private ITrackingSpan CreatePrimaryPrompt()
         {
-            return CreatePrompt(_evaluator.GetPrompt(), ReplSpanKind.Prompt, LastLineNumber);
+            return CreateTrackingSpan(_promptBuffer, _evaluator.GetPrompt(), ReplSpanKind.Prompt);
         }
 
-        private ReplSpan CreatePrompt(string prompt, ReplSpanKind promptKind, int lineNumber)
+        private ITrackingSpan CreateSecondaryPrompt()
         {
-            Debug.Assert(promptKind.IsPrompt());
-            return new ReplSpan(prompt, promptKind, lineNumber);
+            // TODO (crwilcox) format prompt used to get a blank here but now gets "> " from get prompt.
+            return CreateTrackingSpan(_secondaryPromptBuffer, _evaluator.GetPrompt(), ReplSpanKind.SecondaryPrompt);
         }
 
-        private ReplSpan CreateSecondaryPrompt(int lineNumber)
+        private static ITrackingSpan CreateTrackingSpan(ITextBuffer buffer, string textToAppend, ReplSpanKind kind)
         {
-            // TODO (crwilcox) format prompt used to get a blank here but now gets "> " from get prompt.
-            return CreatePrompt(_evaluator.GetPrompt(), ReplSpanKind.SecondaryPrompt, lineNumber);
+            using (var edit = buffer.CreateEdit())
```

Comment:
Does this mean we're putting all the prompts (of each kind) in a single subject buffer, rather than having a separate one for each?